src/api/dates.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Returns whether the given API date is unambiguously in the future, relative
+ * to "now". (start_date and end_date in the API can represent a range of dates
+ * rather than just a single one.)
+ *
+ * The [now] param will be interpreted in local time.
+ */
+export function isInFuture(apiDate: string, now: Date): boolean {
```

Comment:
sheesh! they really need to get quarters/halves in the spec... looks good, though